### ISSUE TITLE
ytree + Gadget ?

### ISSUE THREAD

**Issue by cspotz** [ytree-project/ytree#136]
I have been trying to load data outputed from Gadget 4 to ytree but did not succeed, is YT still compatible with Gadget data ?

I tried 

```python
ytree.load("/fof_subhalo_tab_000.hdf5")
ytree.load("/fof_subhalo_tab_000.hdf5")
ytree.load("/fof_subhalo_tab_000.hdf5")
```
but nothing worked out

Here are my data if you want to have a try.
[ytree.zip](https://github.com/ytree-project/ytree/files/8856679/ytree.zip)

**Version Information**

  * Operating System: linus ubuntu
  * Python Version: 3.7
  * ytree version: 3.1.3

 pip

**Comment by manodeep** [comment]
Not sure that ytree currently supports the  Gadget4 mergertree file format. Regardless, if you are trying to read in mergertrees, you probably have to run the additional postprocessing step of creating the trees by running Gadget4 with restartflag of 8 (source https://wwwmpa.mpa-garching.mpg.de/gadget4/09_special_modules/, section on Mergertrees). 

Disregard if you are trying to load the FOF-subhalo groups (rather than mergertrees) - in that case, you will probably need yt or something similar.

**Comment by cspotz** [comment]
Yes, I run the command you mentionned and also tried to load the outputs:
```python
ytree.load("/tree.hdf5")
ytree.load("/fof_subhalo_tab_000.hdf5")
ytree.load("/subhalo_treelink_018.hdf5")
```
but it did not workout, my guess is that ytree is not currently compatible with the Gadget output.

**Comment by brittonsmith** [comment]
Manodeep is correct, ytree does not currently support this format. If you could share a small sample dataset and any documentation you know of describing the format, I'd be happy to take a look at it.

**Comment by brittonsmith** [comment]
Great, thank you. I'm currently fairly busy, so it might not be until the end of the summer when I can get to this. That said, it looks fairly straightforward, so maybe I can get to it sooner.

**Comment by brittonsmith** [comment]
@cspotz I've just issue PR #138 with a new frontend that works with the data you posted. I did this on the plane yesterday and haven't had time to update the docs yet. Also, do you have any sample data that consists of multiple files? If so, I will try to update the PR to support that. I'm happy for it to be up to a few hundred MB.

**Comment by cspotz** [comment]
Hey @brittonsmith , thanks a lot for the frontend, I am also almost travelling, but try these files.
[treedata.zip](https://github.com/ytree-project/ytree/files/9071291/treedata.zip)


I tried checking for the file tree.zip I provided you and failed. Did you already push your version of ytree ?

```python
import yt
import ytree
print (ytree.__version__)
ytree.load("test.hdf5")
```

Outcome
```
3.2.dev3

---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
/tmp/ipykernel_32052/2375386421.py in <module>
----> 1 ytree.load("test.hdf5")

~/PostDocStrasbourg/NGWDM/trees/ytree/ytree/data_structures/load.py in load(filename, method, **kwargs)
     87 
     88         if len(candidates) == 0:
---> 89             raise IOError(f"Could not determine arbor type for {filename}.")
     90         elif len(candidates) > 1:
     91             errmsg = "Could not distinguish between these arbor types:\n"

OSError: Could not determine arbor type for test.hdf5.


```

Or maybe I am doing something wrong on my side ?

**Comment by brittonsmith** [comment]
@cspotz, the relevant pull request has now been merged into the main repository. If you update your source installation, you'll have it. Please let us know if you encounter any issues. We'll do another release eventually, but probably not until the fall.
